modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/events/SegmentsServicePreAction.java

```diff
@@ -159,29 +160,21 @@ protected void doRun(
 		}
 		else {
 			try {
-				segmentsEntryIds =
-					_segmentsEntrySessionCache.getSegmentsEntryIds(groupId);
-
 				Context context = _requestContextMapper.map(httpServletRequest);
 
-				if (segmentsEntryIds == null) {
-					segmentsEntryIds =
-						_segmentsEntryProviderRegistry.getSegmentsEntryIds(
-							groupId, User.class.getName(), userId, context);
-
-					_segmentsEntrySessionCache.putSegmentsEntryIds(
-						groupId, segmentsEntryIds);
-				}
-				else {
-					context.put("segmentsEntryIds", segmentsEntryIds);
+				Optional.ofNullable(
+					_segmentsEntrySessionCache.getSegmentsEntryIds(groupId)
+				).ifPresent(
+					cachedSegmentsEntryIds -> context.put(
```

Comment:
@cgoncas I'm setting the context parameter here and not inside the RequestContextMapper since I wanted to make it very explicit. For 7.3 we want to added inside the RequestContextMapper or to add a new parameter here `SegmentsEntryProviderRegistry.getSegmentsEntryIds` (changing the current API)

Reply:
What are the drawbacks of moving it to the RequestContextMapper?

Reply:
We'll It is not the thing of doing it inside the RequestContextMapper but the use we're doing of the Context object inside of _isMember_. Currently It's a bit tricky to be aware of this parameter "segmentsEntryIds". I wasn't sure which approach would be better:

1. Use the Context object, with a new key for segments (this requires an API change), populated in RequestContextMapper or
2. Add a new parameter in SegmentsEntryProviderRegistry.getSegmentsEntryIds (API change)

So far, I prefer option 1